machine/v1beta1/types_awsprovider.go

```diff
@@ -44,6 +44,12 @@ type AWSMachineProviderConfig struct {
 	// it should use the default of its subnet.
 	// +optional
 	PublicIP *bool `json:"publicIp,omitempty"`
+	// NetworkInterfaceType specifies the type of network interface to be used for the primary
```

Comment:
do we make use of secondary network interfaces?  I don't mind only controlling this one, but I'd like to know.

Reply:
We only support a single network interface within Machine API across all of the core cloud providers, I'm not sure why this was done as this was pre me joining RH